tests/study/test_repository.py

```diff
@@ -1004,3 +1177,71 @@ def test_update_tags(
     # Check that only "Tag1" and "Tag3" are present in the database
     tags = db_session.query(Tag).all()
     assert {tag.label for tag in tags} == {"Tag1", "Tag3"}
+
+
+@pytest.mark.parametrize(
```

Comment:
I really like the diversification here and i think you could do these a bit more for the other tests you added. For instance, i think you always check the pagination with these parameters `page_nb=1, page_size=2` and it could be different